frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -601,11 +602,11 @@ Except when in two columns mode, where this is limited to showing only the previ
     if not Device:isTouchDevice() and Device:hasDPad() then
         table.insert(menu_items.long_press.sub_item_table, {
             text_func = function()
-                return T(_("Rate of movement in content selection: %1"), G_reader_settings:readSetting("highlight_non_touch_factor", 4))
+                return T(_("Rate of movement in content selection: %1"), G_reader_settings:readSetting("highlight_non_touch_factor" or 4))
```

Comment:
Our advices were quite confusing :/ but this is wrong:
```lua
G_reader_settings:readSetting(
  "highlight_non_touch_factor" or 4   -- this will always be "highlight_non_touch_factor"
)
```
The `or 4` should be outside:
`G_reader_settings:readSetting("highlight_non_touch_factor")      or 4`
Then, use parens wrapping this if needed - but it is not needed here.
`return T(_("Rate of movement in content selection: %1"), G_reader_settings:readSetting("highlight_non_touch_factor") or 4)`

Reply:
> Our advices were quite confusing

you think? I should have just left this alone. It was fine before.

Reply:
is it JUST this one, or do any other cases need changing?

Reply:
**Not** just this one, but all those that have:
`G_reader_settings:readSetting(   "some_string" or some_number   )`
which should be:
`G_reader_settings:readSetting("some_string")    or some_number`

Reply:
then why on earth would @hius07 say 

> This and next changes are wrong.

and then @Frenzie 

> The or 1 should be inside.

Reply:
Because they were busy, often lack precision, and did not care taking you by your little sticky hand like I did ? :)
I guess the mentionned wrongness was about the need for the NC() plural thing.

Reply:
Because adding `or 1` in this piece of code was wrong in logic
https://github.com/koreader/koreader/blob/ffb8cc53d5abf2c7a51dab12ab60575b7f8b49f1/frontend/apps/reader/modules/readerhighlight.lua#L639-L645

Reply:
that is how you should have phrased it then, "This and next changes are wrong." is a great example of a classically ambiguous english phrase. 

thanks @poire-z, it was lovely holding your hand. ;) anyway, all good here now?